### ISSUE TITLE
Kernel panic with undefined behavior message when setting resolution

### ISSUE THREAD

**Issue by arthurmco** [SerenityOS/serenity#12970]
In the "Display Settings" page, setting the screen resolution results in a kernel panic due to undefined behavior

The last messages before the crash were:

```
5.506 DisplaySettings(38:38): Started thread "", tid = 39
22.003 DisplaySettings(38:38): Saving wallpaper path '' to ConfigServer
22.003 FileManager (Desktop)(25:25): Failed to load wallpaper bitmap from path: open: Bad address (errno=14)
22.007 WindowServer(17:17): Updating fonts: 'Katica 10 400 0' and 'Csilla 10 400 0'
22.011 WindowServer(17:17): Closing device /dev/fb0 in preparation for resolution change
22.011 [#0 WindowServer(17:17)]: BochsGraphicsAdapter: resolution set to 1024x768
22.011 [#0 WindowServer(17:17)]: Framebuffer Console: taking 3145728 bytes
22.011 [#0 WindowServer(17:17)]: VC 0: Resized to 128 x 96
22.011 [#0 WindowServer(17:17)]: VC 1: Resized to 128 x 96
22.011 [#0 WindowServer(17:17)]: VC 2: Resized to 128 x 96
22.011 [#0 WindowServer(17:17)]: VC 3: Resized to 128 x 96
22.011 [#0 WindowServer(17:17)]: VC 4: Resized to 128 x 96
22.011 [#0 WindowServer(17:17)]: VC 5: Resized to 128 x 96
[WindowServer(17:17)]: KUBSAN: member access within misaligned address 0xaaaaaaaa of type 'struct KString'
[WindowServer(17:17)]: KUBSAN: at ../../Kernel/KString.cpp, line 67, column: 95
[WindowServer(17:17)]: Kernel + 0x00ff9b70  print_location +0x180
[WindowServer(17:17)]: Kernel + 0x00ffcbff  __ubsan_handle_type_mismatch_v1 +0x1af
[WindowServer(17:17)]: Kernel + 0x00a41599  Kernel::KString::operator delete(void*) +0x49
[WindowServer(17:17)]: Kernel + 0x00e16942  Kernel::Process::sys$mmap(AK::Userspace<Kernel::Syscall::SC_mmap_params const*>) +0x2892
[WindowServer(17:17)]: Kernel + 0x00d5d9a3  syscall_handler +0xea3
[WindowServer(17:17)]: Kernel + 0x00d5caf1  syscall_asm_entry +0x31
[WindowServer(17:17)]: UB is configured to be deadly, halting the system.
```

I can confirm that the resolution changes, but the kernel panic occurs right after that

The screen looks like this (I am using QEMU)
![image](https://user-images.githubusercontent.com/13368343/157518377-53427b56-e525-48a9-a277-c2fea4a95a2d.png)

I checked with the current master (commit `a13079f7`) and this bug still happens

**Comment by kleinesfilmroellchen** [comment]
This is related to #12977, i.e. two different errors happen with KUBSAN and without.

**Comment by supercomputer7** [comment]
I'm working on a fix for this in #12181 now, so I'll upload soon a new commit (or maybe two) to address this issue.
